src/ValidationContext.php

```diff
@@ -176,16 +176,16 @@ public function setParameter(string $key, mixed $value): self
 
     public function isAttributeMissing(): bool
     {
-        return $this->attribute !== null && $this->dataSet !== null && !$this->dataSet->hasAttribute($this->attribute);
+        return $this->attribute !== null && !$this->getDataSet()->hasAttribute($this->attribute);
```

Comment:
It's throw exception when not set data set. Is it purpose of change?

Reply:
Yes. Because a data set is always set here for every context:

https://github.com/yiisoft/validator/blob/d5bc19b6939bdcc7907287ec6c49947a906f2244/src/Validator.php#L85

The lack of it is a programmer error.